packages/editor/src/plugins/drag-drop-bg/index.ts

```diff
@@ -0,0 +1,71 @@
+import { EditorPluginType } from '@editor/types/editor-plugin-type'
+
+import { DragDropBgEditor } from './editor'
+import {
+  type EditorPlugin,
+  type EditorPluginProps,
+  object,
+  string,
+  child,
+  number,
+  boolean,
+  list,
+} from '../../plugin'
+
+export const answerZoneImageProps = child({ plugin: EditorPluginType.Image })
+export const answerZoneTextProps = child({ plugin: EditorPluginType.Text })
+
+export const positionData = object({
+  top: number(0),
+  left: number(0),
+})
+
+export const layoutData = object({
+  width: number(0),
+  height: number(0),
+  visible: boolean(true),
+  lockedAspectRatio: boolean(true),
+})
+
+export const answerData = object({
+  image: answerZoneImageProps,
+  text: answerZoneTextProps,
+})
+
+export const answerZoneProps = object({
+  id: string(''),
+  position: positionData,
+  layout: layoutData,
+  answer: answerData,
+})
+
+export const wrongAnswerProps = object({
+  id: string(''),
+  answer: answerData,
+})
+
+const dragDropBgState = object({
+  answerZones: list(answerZoneProps, 0),
+  backgroundType: string(''),
```

Comment:
string is fine here, but I'd prefer an enum for the options so I can easily see what types are supported.

Reply:
Added enums for backgroundType and backgroundShape, thanks for the suggestion 👍